### PR TITLE
feat(hcpctl): add taxonomy classification to snapshot analyze

### PR DESCRIPTION
**Pull request by gmfrasca**
[ARO-28427](https://issues.redhat.com/browse/ARO-28427)

### What

Adds taxonomy classification to `hcpctl snapshot analyze` output. Each analysis now includes an L1 category, optional L2 subcategory (for Product Failures), and a confidence score assigned by the LLM.

### Why

Automated classification enables downstream aggregation and triage of test failures by category (Azure Problems, Deployment Failures, Product Failures, Test Reliability) without manual review of each analysis.

### Testing

**Unit tests** (`classification_test.go`): schema parsing (with/without classification), validation (14 subtests covering all L1/L2/confidence rules and boundary values), and markdown rendering (with L2, without L2, without classification).

**Manual verification** with `hcpctl snapshot analyze`:
1. Build hcpctl:
```
cd tooling/hcpctl && make build
```

2. Gather a snapshot from a failed Prow job:
```
hcpctl snapshot from-prow-job --url <prow-job-url> --test <test-name> --output-dir <snapshot-path>
```

3. Run the analysis:
```
hcpctl snapshot analyze <snapshot-path>/path/to/<test-name> \
--aro-hcp <path-to-aro-hcp-repo> \
--hypershift <path-to-hypershift-repo> \
--maestro <path-to-maestro-repo> \
--clusters-service <path-to-CS-repo>
```

4. Verify `analysis.json` contains a `classification` object with `l1_category`, `confidence`, and `l2_subcategory` (when applicable)
5. Verify `analysis.md` contains a "Classification" section with Category, Component (if Product Failures), and Confidence percentage

### PR Checklist

- [x] PR is scoped to a single task (no mixed concerns)
- [x] Title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) format
- [x] Summary explains the "Why" behind the change
- [x] Linked to relevant ticket/issue
- [ ] Screenshots included (if graph/UI/metrics changes)
- [x] Self-reviewed the diff
- [ ] CI/CD checks are passing (ignore Tide)
- [ ] Draft PR used for WIP (if applicable)
- [ ] Commit history is clean (rebased/squashed)
- [x] Tricky code blocks are commented
- [ ] Specific reviewers tagged
- [ ] All comment threads resolved before merge

### CODE REVIEW

File: tooling/hcpctl/pkg/agent/prompts/exemplars/clusters-service-worker-race-condition.md [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 A race condition between two independent Clusters Service workers caused the HyperShift NodePool `spec.replicas` to be reverted from `3` back to `2`, even though the CS database correctly stored `replicas=3`. The node pool worker and the upgrade policy worker both performed concurrent read-modify-write operations on the same ManifestWork, and the last writer (upgrade policy worker) overwrote the replica change applied by the first writer (node pool worker).
 
+## Classification
+
+- **Category:** Product Failures
+- **Component:** Cluster Service
+
```
[the remark this answers is not in the document]
Comment (gmfrasca) [reply tooling/hcpctl/pkg/agent/prompts/exemplars/clusters-service-worker-race-condition.md:11]: Intentional - confidence is omitted from exemplars so the LLM infers it from the evidence rather than mimicking a hardcoded value. The system prompt and JSON schema example already demonstrate the confidence field; the exemplars teach format and category assignment, not specific confidence values.

File: tooling/hcpctl/pkg/agent/prompts/exemplars/cluster-installation-ignition-failure.md [hunk 2]
```diff
@@ -4,6 +4,11 @@
 
 The test failed because a router Deployment rollout (triggered by the control-plane-operator adding the ignition-server backend to the `router` ConfigMap) could not complete: the new ReplicaSet's pod was stuck in `FailedCreatePodSandBox` due to Azure CNS errors (`network is not ready - mtpnc is not ready`). The old router pods remained running but served the old haproxy config without the ignition backend, so ignition requests fell through to the kube-apiserver via `default_backend kube_api`, causing a TLS certificate mismatch that left worker VMs stuck in an ignition fetch retry loop until Azure reported `OSProvisioningTimedOut`.
 
+## Classification
+
+- **Category:** Product Failures
+- **Component:** HyperShift
+
```
[the remark this answers is not in the document]
Comment (gmfrasca) [reply tooling/hcpctl/pkg/agent/prompts/exemplars/cluster-installation-ignition-failure.md:11]: Intentional - confidence is omitted from exemplars so the LLM infers it from the evidence rather than mimicking a hardcoded value. The system prompt and JSON schema example already demonstrate the confidence field; the exemplars teach format and category assignment, not specific confidence values.

File: tooling/hcpctl/pkg/agent/prompts/exemplars/cluster-installation-azure-disk-failure.md [hunk 3]
```diff
@@ -5,6 +5,10 @@ This document shows a proof chain for a failure to install a cluster coming from
 An ARO HCP cluster failed to install because an Azure Disk could not be mounted for an `etcd` pod in the hosted control
 plane due to internal Azure Disk errors.
 
+## Classification
+
+- **Category:** Azure Problems
+
```
[the remark this answers is not in the document]
Comment (gmfrasca) [reply tooling/hcpctl/pkg/agent/prompts/exemplars/cluster-installation-azure-disk-failure.md:11]: Intentional - confidence is omitted from exemplars so the LLM infers it from the evidence rather than mimicking a hardcoded value. The system prompt and JSON schema example already demonstrate the confidence field; the exemplars teach format and category assignment, not specific confidence values.

File: tooling/hcpctl/pkg/agent/prompts/exemplars/cluster-creation-timeout-z-stream-upgrade.md [hunk 4]
```diff
@@ -9,6 +9,11 @@ backend kept reporting *"hosted cluster has no installed version"* through the t
 New kube-apiserver ReplicaSet `54b7f655f6` pods logged `etcd-client` dial failures while etcd quorum stayed healthy
 (`EtcdAvailable=True`).
 
+## Classification
+
+- **Category:** Product Failures
+- **Component:** Backend
+
```
[the remark this answers is not in the document]
Comment (gmfrasca) [reply tooling/hcpctl/pkg/agent/prompts/exemplars/cluster-creation-timeout-z-stream-upgrade.md:16]: Intentional - confidence is omitted from exemplars so the LLM infers it from the evidence rather than mimicking a hardcoded value. The system prompt and JSON schema example already demonstrate the confidence field; the exemplars teach format and category assignment, not specific confidence values.

File: tooling/hcpctl/pkg/agent/classification_test.go [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+		Summary:        "summary",
+		Classification: c,
+		Chain: []ChainLink{
+			{
+				Question: FirstChainQuestion,
+				Answer:   "answer",
+				Proof: []ProofItem{
+					{Type: "log", Source: "error", Lines: [2]int{1, 1}},
+					{Type: "code", Repo: "ARO-HCP", File: "main.go", Lines: [2]int{1, 1}},
+				},
```
[the remark this answers is not in the document]
Comment (gmfrasca) [reply tooling/hcpctl/pkg/agent/classification_test.go:92]: Changed `newValidationContext` to use `t.TempDir()` and create a real `main.go` stub file, so the code proof validation passes cleanly and the "valid" classification cases now produce zero total problems (not just zero classification problems).

**Review by galchammat — COMMENTED** [review]
The logic looks sound. Approver might want to specifically look at the L1 and L2 categories defined below [this comment](https://github.com/Azure/ARO-HCP/pull/6504/changes#diff-dbeaedffc08449710cd6d60069d8a0e7f0333fc734e9267c36b513ffd2a0f85dR23) and explicitly defined in the [JIRA](https://issues.redhat.com/browse/ARO-28427).

**Review by janboll — COMMENTED** [review]
/lgtm
/approve

**Comment by stevekuznetsov** [comment]
@gmfrasca @janboll this kind of stuff needs my eyes, please.
